apps/verticalhome/js/more_apps_navigation.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+(function (exports) {
+
+  var MoreAppsNavigation = {
+    controls: null,
+    selectedElemIndex: 0,
+    selectElement: null,
+
+    init: function () {
+      this.controls = app.grid.getItems();
+      this.reset();
+    },
+
+    reset: function () {
+      if (this.selectElement) {
+        this.selectElement.classList.remove('selected');
+      }
+      this.selectedElemIndex = 0;
+      this.selectElement = this.controls[this.selectedElemIndex].element;
+      this.selectElement.classList.add('selected');
+      this.clearNavigationEvents();
+      window.addEventListener('keydown', this);
+    },
+
+    clearNavigationEvents: function () {
+      window.removeEventListener('keydown', this);
+    },
+
+    handleEvent: function (e) {
+      var deltaIndex = 0;
+      e.preventDefault();
+      switch (e.key) {
+        case 'ArrowUp':
+          deltaIndex = -2;
+          this.changeSelectElem(deltaIndex);
+          if (!this.isVisible(this.selectElement)) {
+            this.selectElement.scrollIntoView({behavior:"smooth", block: "start"});
+          }
+          break;
+        case 'ArrowDown':
+          deltaIndex = 2;
+          this.changeSelectElem(deltaIndex);
+          if (!this.isVisible(this.selectElement)) {
+            this.selectElement.childNodes[0].scrollIntoView({behavior:"smooth", block: "end"});
+          }
+          break;
+        case 'ArrowLeft':
+          deltaIndex = -1;
+          this.changeSelectElem(deltaIndex);
+          if (!this.isVisible(this.selectElement)) {
+            this.selectElement.scrollIntoView({behavior:"smooth", block: "start"});
+          }
+          break;
+        case 'ArrowRight':
+          deltaIndex = 1;
+          this.changeSelectElem(deltaIndex);
+          if (!this.isVisible(this.selectElement)) {
+            this.selectElement.childNodes[0].scrollIntoView({behavior:"smooth", block: "end"});
+          }
```

Comment:
There is a lot of duplicated code for these four arrow keys. Can you write a utility function that just takes the deltaIndex and does the right thing?  I'd also like to see a comment that explains why you sometimes have to use childNodes[0] and sometimes do not have to.